src/accelerate/utils/imports.py

```diff
@@ -425,6 +425,22 @@ def is_xpu_available(check_device=False):
     return hasattr(torch, "xpu") and torch.xpu.is_available()
 
 
+def get_xpu_available_memory(device_index: int):
+    if is_ipex_available():
+        try:
+            from intel_extension_for_pytorch.xpu import mem_get_info
+
+            return mem_get_info(device_index)[0]
+        except Exception:
+            warnings.warn(
+                "The XPU `mem_get_info` API is available in IPEX version >=2.5. The current returned availabel memory is incorrect. Pls consider upgrading your IPEX version."
+            )
+            return torch.xpu.max_memory_allocated(device_index)
+    else:
+        # this branch is for stock PyTorch 2.4
+        return torch.xpu.max_memory_allocated(device_index)
```

Comment:
Let's move this to `utils.memory` instead of imports.

For the implementation itself, we can do:

```python
if is_ipex_available():
    ipex_version = importlib.metadata.version("intel_extension_for_pytorch")
    if compare_versions(ipex_version, ">=", "2.5"):
        from intel_extension_for_pytorch.xpu import mem_get_info
        return mem_get_info(device_index)[0]
    warnings.warn(
        "The XPU `mem_get_info` API is available in IPEX version >=2.5. The current returned available memory is incorrect. Please consider upgrading your IPEX version."
    )
return torch.xpu.max_memory_allocated(device_index)
```

Reply:
thanks for the nice refactoring! code updated